src/hooks/useNotificationManager.tsx

```diff
@@ -0,0 +1,147 @@
+import React, {
+  createContext,
+  useCallback,
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import { AppState, AppStateStatus } from 'react-native';
+import {
+  PrivatePostNotification,
+  useNotifications,
+  Notification,
+} from './useNotifications';
+import { useAsyncStorage } from './useAsyncStorage';
+import { onNotificationReceived } from '../common';
+import { compact } from 'lodash';
+
+export interface NotificationsManagerContextType {
+  unreadCount: number | undefined;
+  setNotificationsRead: () => void;
+  /**
+   * undefined - async storage has not been loaded yet
+   * null - async storage has been loaded but no value
+   * Date - async storage has loaded and key was set
+   */
+  lastReadAt: Date | null | undefined;
+  privatePostsUserIds: string[] | undefined;
+}
+
+export const NotificationsManagerContext =
+  createContext<NotificationsManagerContextType>({
+    unreadCount: 0,
+    privatePostsUserIds: [],
+    setNotificationsRead: () => {},
+    lastReadAt: null,
```

Comment:
Should this be `undefined` per your comment above?

Reply:
Yes, indeed